src/main/java/schwarz/jobs/interview/coupon/web/CouponResource.java

```diff
@@ -0,0 +1,69 @@
+package schwarz.jobs.interview.coupon.web;
+
+
+import java.util.List;
+import java.util.NoSuchElementException;
+
+import javax.validation.Valid;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import schwarz.jobs.interview.coupon.core.domain.Coupon;
+import schwarz.jobs.interview.coupon.core.services.CouponService;
+import schwarz.jobs.interview.coupon.core.domain.Basket;
+import schwarz.jobs.interview.coupon.web.dto.CouponApplicationRequestDTO;
+import schwarz.jobs.interview.coupon.web.dto.CouponDTO;
+import schwarz.jobs.interview.coupon.web.dto.CouponRequestDTO;
+
+@Controller
+@RequiredArgsConstructor
+@RequestMapping("/api/coupons")
```

Comment:
Added /coupons to the endpoint to indicate that is is related to the "coupons" resource.